### ISSUE TITLE
\p in in RegExp not working

### ISSUE THREAD

**Issue by atoomic** [CpanelInc/perl-compiler#64]
For example here \p{IsWord} is not matching what it should

```perl
my $x = q[OK]; 
$x =~ s/(\p{IsWord}+)/lc($1)/ge; 
print $x
```

```
> perl -e 'my $x = q[OK]; $x =~ s/(\p{IsWord}+)/lc($1)/ge; print $x'
ok
```

```
> perlcc -r -e 'my $x = q[OK]; $x =~ s/(\p{IsWord}+)/lc($1)/ge; print $x'
OK
```

sample xtestc/0242.t

**Comment by atoomic** [comment]
The problem here is compiling when \p is visible

This works
```perl
my $x = q[HELLO]; 
my $q = q[\p{IsWord}]; # q as quoted string
$x =~ s/($q+)/lc($1)/ge; 

print "# $x\n";
print qq[ok\n] if $x eq 'hello';
```

```
# hello
ok
```

This fails
```perl
my $x = q[HELLO]; 
my $q = qr[\p{IsWord}]; # q as quoted RegExp
$x =~ s/($q+)/lc($1)/ge; 

print "# $x\n";
print qq[ok\n] if $x eq 'hello';
```

```
# HELLO
```

**Comment by atoomic** [comment]
This could be simplified to a simple RegExp

```perl
my $q = qr[\p{IsWord}];
print qq[match ok\n] if q[hello] =~ qr{$q};
```

**Comment by toddr** [comment]
Fixed in 4993d9be79f1b931a73fa8d434d2f0f7597e64ae
